crates/astria-conductor/src/executor/mod.rs

```diff
@@ -396,6 +412,35 @@ impl Executor {
         // TODO(https://github.com/astriaorg/astria/issues/624): add retry logic before failing hard.
         let executable_block = ExecutableBlock::from_sequencer(block, self.state.rollup_id());
 
+        // Stop executing soft blocks at the sequencer stop block height (exclusive). If we are also
+        // executing firm blocks, we let execution continue since one more firm block will be
+        // executed before `execute_firm` initiates a restart. If we are in soft-only mode, we
+        // return a `StopHeightExceded::Sequencer` error to signal a restart.
+        if self.state.sequencer_stop_block_height().value() > 0
+            && executable_block.height >= self.state.sequencer_stop_block_height()
+        {
+            let res = if self.mode.is_with_firm() {
+                info!(
+                    height = %executable_block.height,
+                    "received soft block whose height is greater than or equal to stop block height in {} mode. \
+                    dropping soft block and waiting for next firm block before attempting restart",
+                    self.mode,
+                );
+                Ok(())
+            } else {
+                info!(
+                    height = %executable_block.height,
+                    "received soft block whose height is greater than or equal to stop block \
+                    height in soft only mode. shutting down and attempting restart",
+                );
+                Err(StopHeightExceded::Sequencer).wrap_err(
+                    "soft height met or exceeded sequencer stop height, attempting restart with \
+                     new genesis info",
+                )
+            };
+            return res;
```

Comment:
Don't bind `res` and explicitly return. Just `return Ok` and `return Err` directly.